File: library/src/main/java/org/bottomnav/BottomNavAdapter.java
package org.bottomnav;

import android.databinding.BindingAdapter;
import android.view.Menu;

import java.util.List;

public class BottomNavAdapter {

    @BindingAdapter("setOnClickItem")
    public static void onOnClickItem(BottomNavMenu bottomNavMenu, OnClickItemListener listener) {
        bottomNavMenu.setOnItemClickListener(listener);
    }

    @BindingAdapter("setItems")
    public static void setMenuItems(BottomNavMenu bottomNavMenu, Menu menu) {
        bottomNavMenu.setItems(menu);
    }

    @BindingAdapter("setItems")
    public static void setMenuItems(BottomNavMenu bottomNavMenu, List<BottomNavMenuItem> itemList) {
        bottomNavMenu.setItems(itemList);
    }
}

File: library/src/main/java/org/bottomnav/BottomNavAdapter.kt
package org.bottomnav

import android.databinding.BindingAdapter

object BottomNavAdapter {

    @JvmStatic
    @BindingAdapter("setOnClickItem")
    fun onOnClickItem(bottomNavMenu: BottomNavMenu, listener: OnClickItemListener) {
        bottomNavMenu.addOnItemClickListener(listener)
    }

    @JvmStatic
    @BindingAdapter("setItems")
    fun setItems(bottomNavMenu: BottomNavMenu, itemList: List<BottomNavMenuItem>) {
        bottomNavMenu.setItems(itemList)
    }
}
